oauth2/src/test/groovy/com/okta/spring/boot/oauth/env/OktaEnvironmentPostProcessorApplicationListenerTest.groovy

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.spring.boot.oauth.env
+
+import ch.qos.logback.classic.Level
+import ch.qos.logback.classic.Logger
+import ch.qos.logback.classic.LoggerContext
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.core.read.ListAppender
+import org.slf4j.LoggerFactory
+import org.springframework.boot.context.event.ApplicationPreparedEvent
+import org.springframework.context.ConfigurableApplicationContext
+import org.springframework.core.env.MapPropertySource
+import org.springframework.mock.env.MockEnvironment
+import org.testng.annotations.AfterMethod
+import org.testng.annotations.Test
+
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.is
+import static org.mockito.Mockito.mock
+import static org.mockito.Mockito.when
+
+class OktaEnvironmentPostProcessorApplicationListenerTest {
+
+    private static final String LOGGER_NAME = "com.okta.spring.boot.oauth.env.OktaEnvironmentPostProcessorApplicationListener"
+    private static final String MSG = "Your Okta Issuer URL is missing. You can copy your domain from the Okta Developer Console. " +
+        "Follow these instructions to find it: https://bit.ly/finding-okta-domain" + System.lineSeparator() +
+        "To fix this add the `okta.oauth2.issuer` property to your application environments."
+    private static final Logger log = (Logger) LoggerFactory.getLogger(LOGGER_NAME)
```

Comment:
It still looks like this technique the same threading issues as before.  Two tests could be running at the same time, using the same logger. This may result in one test getting the log messages from the other.

One way around this could be to update your `LogsAppender` to only accept messages from a specific thread.  (though there might be a better way of doing this)

If I'm missing something please let me know.

Reply:
Ok, now I've gotten your point.
Thanks!